perception/autoware_camera_streampetr/src/node.cpp

```diff
@@ -0,0 +1,382 @@
+
+  network_->inference_detector(
+    data_store_->get_image_input(), ego_pose, ego_pose_inv, data_store_->get_image_shape(),
+    data_store_->get_camera_info_vector(),
+    get_camera_extrinsics_vector(data_store_->get_camera_link_names()), prediction_timestamp,
+    output_objects, forward_time_ms);
+
+  if (multithreading_) data_store_->unfreeze_updates();
+
+  double inference_time_ms = -1.0;
+  if (stop_watch_ptr_) inference_time_ms = stop_watch_ptr_->toc("latency/inference", true);
+
+  DetectedObjects output_msg;
+  output_msg.objects = output_objects;
+  output_msg.header.frame_id = "base_link";
+  output_msg.header.stamp = stamp;
+  pub_objects_->publish(output_msg);
+
+  if (debug_publisher_ptr_ && stop_watch_ptr_) {
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/total", stop_watch_ptr_->toc("latency/total", true));
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/preprocess", data_store_->get_preprocess_time_ms());
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference", inference_time_ms);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference/backbone", forward_time_ms[0]);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference/ptshead", forward_time_ms[1]);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference/pos_embed", forward_time_ms[2]);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/inference/postprocess", forward_time_ms[3]);
+    debug_publisher_ptr_->publish<autoware_internal_debug_msgs::msg::Float64Stamped>(
+      "latency/cycle_time_ms", stop_watch_ptr_->toc("latency/cycle_time_ms", true));
+    stop_watch_ptr_->tic("latency/cycle_time_ms");
+  }
+}
+
+std::vector<float> StreamPetrNode::get_camera_extrinsics_vector(
+  const std::vector<std::string> & camera_links)
+{
+  std::vector<float> intrinsics_all = data_store_->get_camera_info_vector();
+
+  std::vector<float> res;
+  res.reserve(camera_links.size() * 16);
+
+  for (size_t i = 0; i < camera_links.size(); ++i) {
+    Eigen::Matrix4f K_4x4 = Eigen::Matrix4f::Identity();
+    {
+      size_t offset = i * 16;
+      for (int row = 0; row < 4; ++row) {
+        for (int col = 0; col < 4; ++col) {
+          K_4x4(row, col) = intrinsics_all[offset + row * 4 + col];
+        }
+      }
+    }
+    geometry_msgs::msg::TransformStamped transform_stamped;
+    try {
+      transform_stamped =
+        tf_buffer_.lookupTransform(camera_links[i], "base_link", tf2::TimePointZero);
+    } catch (const tf2::TransformException & ex) {
+      throw std::runtime_error(
+        "Could not transform from base_link to " + camera_links[i] + ": " + std::string(ex.what()));
```

Comment:
[Q] Should we throw exception here even though catching `tf2::TransformException`?